### PR TITLE
move to puma

### PR DESCRIPTION
**Pull request by mi-wood**
# Context
- Fixes https://github.com/RefugeRestrooms/refugerestrooms/issues/347
- Move to puma as recommended by Heroku

# Summary of Changes

- adds puma gem and changes procfile
- thread env variable has been updated on heroku

### CODE REVIEW

**Comment by tkwidmer** [comment]
This looks good to me, have we deployed in a test environment to make sure it works?

File: app.json [hunk 1]
```diff
@@ -14,18 +14,6 @@
     "EMAIL_PASSWORD": {
       "required": true
     },
-    "HEROKU_POSTGRESQL_AQUA_URL": {
```
Comment (mi-wood) [inline app.json]: @tkwidmer I imagine these are old DB names?

**Comment by DeeDeeG** [comment]
> thread env variable has been updated on heroku

This could be set to "one," to make sure we are de-facto "thread safe," or this could be two or more, for speed.

Which way are we thinking of going with this? Would anyone be interested in trying multiple threads anyway, just to see if everything works? Or play it more safely by sticking with one thread?

 (Issues, if any, would likely be intermittent race conditions, and therefore harder to diagnose than some issues are. Also, since most of our Ruby code is third-party via gems, fixing the issues could involve working upstream with those gem maintainers. But if there are no such issues, it's all moot and we can use however many threads we want.)

**Comment by mi-wood** [comment]
@DeeDeeG I set it to 1 and figure that should be fine. I don't think using one thread is going to be an issue performance wise.

I also created https://github.com/RefugeRestrooms/refugerestrooms/issues/445 to evaluate this for the future.

**Pull request merged**